app/react/Users/components/UserSidePanel.tsx

```diff
@@ -0,0 +1,267 @@
+                {errors.email.type === 'required' && <Translate>Email is required</Translate>}
+                {errors.email.type === 'validate' && <Translate>Duplicated email</Translate>}
+                {errors.email.type === 'pattern' && <Translate>Invalid email</Translate>}
+              </div>
+            )}
+          </div>
+          <div id="role_field" className="form-group nested-selector">
+            <div>
+              <label>
+                <Translate>Role</Translate>
+              </label>
+              <button
+                id="role-info"
+                className="role-info"
+                type="button"
+                onClick={togglePermissionList}
+              >
+                <Icon icon="info-circle" />
+              </button>
+            </div>
+            <select
+              name="role"
+              className="form-control"
+              onChange={handleInputChange}
+              value={userToSave.role}
+            >
+              {Object.values(UserRole).map(role => (
+                <option key={role} value={role}>
+                  {role}
+                </option>
+              ))}
+            </select>
+          </div>
+          <div id="username_field" className="form-group nested-selector">
+            <label>
+              <Translate>Username</Translate>
+            </label>
+            <input
+              type="text"
+              className="form-control"
+              autoComplete="off"
+              value={userToSave.username}
+              name="username"
+              onChange={handleInputChange}
+              ref={register({
+                required: true,
+                validate: isDuplicated,
+                maxLength: 50,
+                minLength: 3,
+              })}
+            />
+            {errors.username && (
+              <div className="validation-error">
+                <Icon icon="exclamation-triangle" size="xs" />
+                {errors.username.type === 'required' && <Translate>Username is required</Translate>}
+                {errors.username.type === 'validate' && <Translate>Duplicated username</Translate>}
+                {errors.username.type === 'maxLength' && (
+                  <Translate>Username is too long</Translate>
+                )}
+                {errors.username.type === 'minLength' && (
+                  <Translate>Username is too short</Translate>
+                )}
+              </div>
+            )}
+          </div>
+          <div id="password_field" className="form-group nested-selector">
+            <label>
+              <Translate>Password</Translate>
+            </label>
+            <input
+              type="password"
+              className="form-control"
+              autoComplete="off"
+              value={userToSave.password}
+              name="password"
+              onChange={handleInputChange}
+              ref={register({ maxLength: 50 })}
```

Comment:
Is there a reason for limiting the pass length?

Reply:
Basically for usability, but I removed it in case the user wants to use an encrypted large password